sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -132,6 +132,26 @@ class EthService(
       .orNull
   }
 
+  @RpcMethod("eth_getBlockTransactionCountByHash")
+  def getBlockTransactionCountByHash(hash: Hash): Quantity = {
+    blockTransactionCount(_ => bytesToId(hash.toBytes))
+  }
+
+  @RpcMethod("eth_getBlockTransactionCountByNumber")
+  def getBlockTransactionCountByNumber(tag: String): Quantity = {
+    blockTransactionCount(nodeView => getBlockIdByTag(nodeView, tag))
+  }
+
+  private def blockTransactionCount(getBlockId: NV => ModifierId): Quantity = {
+    applyOnAccountView { nodeView =>
+      nodeView.history
+        .getStorageBlockById(getBlockId(nodeView))
+        .map(_.transactions.count(_.isInstanceOf[EthereumTransaction]))
```

Comment:
I think it is safe to remove `.count(_.isInstanceOf[EthereumTransaction])` and use `.size`.
I assume that all the transactions are valid Account/Eth Txs